--- src-original/Cargo.toml
+++ src/Cargo.toml
@@ -29,7 +29,10 @@
 foreign-types-shared = "0.1"
 openssl = "0.10.80"
 openssl-sys = "0.9.116"
 pem = { version = "3", default-features = false }
-pyo3 = { version = "0.29", features = ["abi3"] }
+
+# Chaquopy: add abi3-py310 for Android cp311 wheels.
+pyo3 = { version = "0.29", features = ["abi3", "abi3-py310"] }
+
 pyo3-build-config = { version = "0.29" }
 self_cell = "1"
